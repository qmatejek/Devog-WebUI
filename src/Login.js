import React from 'react';
import {Button, Form} from 'react-bootstrap';
import styled from 'styled-components';
import axios from 'axios';


const Container = styled.div`
  width: 100%;
  max-width: 330px;
  padding: 15px;
  margin: auto;
  height: 100%;
  padding-top: 100px;

`




export class Login extends React.Component {
    constructor(props) {
        super(props);
        
        this.state = {
            username:'',
            password:'',
            errorMessage: '',
        }
        
        this.submitHandler = this.submitHandler.bind(this);
    }

    

    changeHandler = e => {
        this.setState({[e.target.name]: e.target.value})
    }

    submitHandler(e){

        e.preventDefault()


        //Makes POST request to the API.
        axios.post('http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/login/', this.state)
        .then(Response => {

            //Stores token for use with other components
            localStorage.setItem('token', Response.data.token)

            //Stores username for use with other components
            if(this.state.username != null){
                localStorage.setItem('username', this.state.username)
            }

            this.props.history.push('/') //Sends user to the home page.
            window.location.reload(false); //Forces refresh to reflect change in login status. (login/logout button)

        })
        .catch(error => {
            this.setState({errorMessage: error.message})
        })
    }

    render() {
        const {username, password} = this.state

        //I know this error handling for incorrect login isn't the best, but we do intend to fix it.
        if(this.state.errorMessage !== ''){
            return(
                <div>
                    <br/><br/>
                    <h1>Oops! An error has occured!</h1>
                    <h3>Error: {this.state.errorMessage}</h3>
                    <h4>Are you sure you entered the correct login credentials?</h4>
                    <h5>Please refresh page and retry</h5>
                </div>
            );
        }
        else{
            return (
                <Container data-testid="login-form-container">

                    <Form className="login-form" data-testid="login-form" onSubmit={this.submitHandler}>
                        <h2 className="text-center">Sign in</h2>

                        <Form.Group controlId="formBasicEmail">
                            <Form.Label>Username</Form.Label>
                            <Form.Control data-testid="username-input" type="text" 
                            placeholder="Enter username" name="username" value = {username} onChange={this.changeHandler}/>
                        </Form.Group>

                        <Form.Group controlId="formBasicPassword">
                            <Form.Label>Password</Form.Label>
                            <Form.Control data-testid="password-input" type="password" 
                            placeholder="Password" name="password" value = {password} onChange={this.changeHandler}/>
                        </Form.Group>

                        <Form.Group controlId="formBasicCheckbox">
                            <Form.Check type="checkbox" label="Remember me" />
                        </Form.Group>

                        <Button data-testid="login-submit" variant="dark" type="submit" size="lg" block>
                            Log in
                        </Button>
                        
                    </Form>

                </Container>  
                        
            );
        }

        
    }
}

export default Login;