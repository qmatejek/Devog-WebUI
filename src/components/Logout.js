import React from 'react';
import axios from 'axios';

export class Logout extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      errorMessage: '',
      token: localStorage['token'],
      auth: `Token ${localStorage['token']}`,
    };
  }

  componentDidMount() {
    this.setState({
      token: localStorage['token'],
      auth: `Token ${localStorage['token']}`
    });
  }
  
  performLogout() {

    const config = {
        url: 'http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/logout/',
        method: 'post',
        headers: {"Authorization":`${this.state.auth}`, 'Content-Type':'application/json'}
    }
      
    axios.request(config)
        .then(() => {
            this.removeItemsFromLocalStorage();
            window.location.replace("/");
        })
        .catch(error => {
            this.setState({ errorMessage: error.message });
        });
                
  }
  
  removeItemsFromLocalStorage() {
    localStorage.removeItem('token')
    localStorage.removeItem('username')
  }
  
  render() {
    if(this.state.errorMessage !== '') {
      return <div>
        <br/><br/>
        <h1>Oops! An error has occured!</h1>
        <h3>Error: {this.state.errorMessage}</h3>
      </div>;
    }
    else{
        return <React.Fragment>
        <span className="navbar-button">
          <button className="btn dark"
            onClick={ () => this.performLogout() }>logout</button>
        </span>
      </React.Fragment>;        
    }

    
  }
}

export default Logout;