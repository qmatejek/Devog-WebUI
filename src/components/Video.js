import React, { Component } from 'react';
import axios from 'axios';
import ReactPlayer from 'react-player';

export class Video extends Component {

    constructor(props) {
        super(props);
        
        this.state = {
            token: localStorage.getItem('token'),
            auth: 'Token ' + localStorage.getItem('token'),
            videos: [],
            streams: [],
            streamName: '',
            streamURL: '',
            errorMessage: '',
        }
        
        this.fetchUp = React.createRef();
        this.fetchLv = React.createRef();
    }

    
    //Handles changes made to the text fields for adding a livestream
    changeHandler = e => {
        this.setState({[e.target.name]: e.target.value})
    }



    //Handles fetching the uploaded videos once the "Fetch Videos" button has been pressed.
    fetchUpSubmitHandler(e){
        e.preventDefault()


        let user = localStorage.getItem('username');

        let url = 'http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/' + user;

        axios.get(url, {headers: {"authorization":this.state.auth}})
        .then(Response => {

            Response.data.map( vid => (
                this.setState({videos: this.state.videos.concat([vid])})
            ))
            
        })
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

        this.fetchUp.current.setAttribute("disabled", "disabled");
        this.fetchLv.current.setAttribute("disabled", "disabled");
    }

    //Handles fetching the live streams once the "Fetch Livestreams" button has been pressed.
    fetchLvSubmitHandler(e){
        e.preventDefault()

        let user = localStorage.getItem('username');
        let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/" + user + "/live";

        axios.get(url, {headers: {"authorization":this.state.auth}})
        .then(Response => {

            Response.data.map( strm => (
                this.setState({streams: this.state.streams.concat([strm])})
            ))
            
        })
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

        this.fetchUp.current.setAttribute("disabled", "disabled");
        this.fetchLv.current.setAttribute("disabled", "disabled");

    }
    
    //Handles the pressing of the edit video submit button for each video. "id" is the id of the video to be edited.
    editSubmitHandler(e, id){
        e.preventDefault()

        let user = localStorage.getItem('username')

        let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/" + user + "/" + id;

        axios.put(url, {"name":e.target[0].value}, {headers: {"Authorization":this.state.auth}} )
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

        window.location.reload(false);
    }

    //Not currently working due to backend issue on sponsor's side... "id" is the id of the live stream to be edited.
    editLiveSubmitHandler(e, id){
        e.preventDefault()

        let un = localStorage.getItem('username')

        let url = 'http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/' + un + '/live/' + id;

        axios.put(url, {"name":e.target[0].value}, {headers: {"Authorization":this.state.auth}} )
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

    }

    //Handles the pressing of the "Upload Video" button.
    fileButtonHandler(e){
        e.preventDefault()

        let user = localStorage.getItem('username')

        let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/" + user;
        const headers = {headers: {"Authorization":this.state.auth, 'Content-Type':'application/json'}};

        const file = e.target.files[0];
        const fileName = e.target.files[0].name;

        const formData = new FormData();
        formData.append('name',fileName);
        formData.append('location',file)

        axios.post(url, formData, headers)
        .then(Response => {
            alert('Successfully uploaded: ' + fileName  + '\nYou may need to refresh page to see new video.')
        })
        .catch(error => {
            this.setState({errorMessage: error.message})
        })
    }

    //Handles the pressing of the "delete" button for each uploaded video. "id" is the id of the video to be deleted.
    delSubmitHandler(e, id){
        e.preventDefault()

        let user = localStorage.getItem('username')

        let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/" + user + "/" + id; 

        axios.delete(url, {headers: {"Authorization":this.state.auth}} )
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

        window.location.reload(false);
    }

    //Handlees the pressing of the "delete" button for each live stream. "id" is the id of the stream to be deleted.
    delLiveSubmitHandler(e, id){
        e.preventDefault()

        let un = localStorage.getItem('username')

        let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/" + un + "/live/" + id;

        axios.delete(url, {headers: {"Authorization":this.state.auth}} )
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

        window.location.reload(false);
    }

    //Handles the pressing of the "Add Livestream" button.
    addLiveSubmitHandler(e){
        e.preventDefault()

        let user = localStorage.getItem('username')

        let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/" + user + "/live";

        const headers = {headers: {"Authorization":this.state.auth, 'Content-Type':'application/json'}};

        const formData = new FormData();
        formData.append('name',this.state.streamName);
        formData.append('link',this.state.streamURL)

        axios.post(url, formData, headers)
        .then(Response => {
            alert('Successfully added: ' + this.state.streamName + '\nYou may need to refresh page to see new stream.')
        })
        .catch(error => {
            this.setState({errorMessage: error.message})
        })

    }


    render() {

        const subStyle = {
            "textShadow": "2px 2px 5px black",
            "paddingTop":"5px",
            "paddingBottom": "0px",
            "paddingLeft":"95px",
        };

        const UploadStyle = {
            "paddingTop":"5px",
            "paddingBottom": "5px",
            "paddingLeft":"5px",
        };

        const listStyle = {
            "listStyleType": "none",
            "display": "inline-block",
            "padding": "40px",
        };

        const footerStyle = {
            "textShadow": "2px 2px 5px black",
            "paddingTop":"40px",
            "paddingBottom": "40px",
            "backgroundColor": "#449342",
            "paddingLeft": "20px",
        };

        const formStyle = {
            "padding":"5px",
            "display":"inline",
        };

        const addLiveStyle = {
            "textShadow": "2px 2px 5px black",
            "paddingTop":"5px",
            "paddingBottom": "5px",
            "paddingRight":"5px",
        };

        const formDivStyle = {
            "paddingTop":"10px",
            "paddingBottom":"10px",
        };


        if(this.state.errorMessage !== ''){
            return(
                <div>
                    {console.log("error branch")}
                    <br/><br/>
                    <h1>Oops! An error has occured!</h1>
                    <h3>Error: {this.state.errorMessage}</h3>
                </div>
            );
        }
        else if(this.state.token != null && this.state.errorMessage === '')
        {
            return (
                <center>
                    <div className="fetchButton">
                            <form style={formStyle} onSubmit={(e) => this.fetchUpSubmitHandler(e)}>
                                <button ref={ this.fetchUp } className="btn dark" type="submit">Fetch Videos</button>
                            </form>
                        
                            <form onSubmit={(e) => this.fetchLvSubmitHandler(e)} style={formStyle}>
                                <button ref={ this.fetchLv } className="btn dark" type="submit">Fetch Livestreams</button>
                            </form>

                    </div>

                    <div style={formDivStyle}>
                        <input style={UploadStyle} type="file" name="file" onChange={(e) => this.fileButtonHandler(e)} />

                        <form style={addLiveStyle} onSubmit={(e) => this.addLiveSubmitHandler(e)} >
                            <input className="botPad5" type="text" name="streamName" placeholder="Enter Stream Name" onChange={this.changeHandler}/> 
                            <input className="botPad5" type="text" name="streamURL" placeholder="Enter Stream Url" onChange={this.changeHandler}/>
                            <button className="btn dark margin5" type="submit">Add Livestream</button>
                        </form>
                    </div>


                    <div className="vidBackground">
                        <React.Fragment>

                            {/* This is what is displayed if the "Fetch Videos" button is pressed. */}
                            {this.state.videos.map(vid => (

                                <ul style={listStyle} key={vid.id}>
                                    <li key={vid.id}>

                                        <video src={vid.location} width="480" height="270" controls>
                                            Your browser does not support the video tag.
                                        </video>

                                        {/* This displays the current title of the video and the edit button */}
                                        <form className="inline"  onSubmit={(e) => {this.editSubmitHandler(e, vid.id )}}>
                                            <h3> Videos current title: {vid.name} </h3>
                                            <input className = "botPad5" type="text" placeholder="New title" id="title" name="title"/>
                                            <button className="btn dark margin5" type="submit">Submit</button>
                                        </form>

                                        <form className="inline"  onSubmit={(e) => {this.delSubmitHandler(e, vid.id )}}>
                                            <button className="btn dark" type="submit">Delete Video</button>
                                        </form>


                                    </li>
                                </ul>
                            ))

                            }

                            {/* This is what is displayed if the "Fetch Livestreams" button is pressed. */}
                            {this.state.streams.map(strm => (
                                <ul style={listStyle}>
                                    <li key={strm.id}>
                                        {/* <ReactPlayer url={strm.link} controls/> */}
                                        {/* <video src={strm.link} width="550" height="350" controls>
                                            Your browser does not support the video tag.
                                        </video> 
                                        Still trying to figure out the best one to use...*/}

                                        <iframe title="streams" width="480" height="270" src={strm.link}></iframe>

                                        {/* This displays the current title of the stream and the edit button. */}
                                        <form className="inline" onSubmit={(e) => {this.editLiveSubmitHandler(e, strm.id )}}>
                                            <h3> Videos current title: {strm.name} </h3>
                                            <input type="text" placeholder="New title" id="title" name="title"/>
                                            <button className="btn dark margin5" type="submit">Submit</button>
                                        </form>

                                        <form className="inline" onSubmit={(e) => {this.delLiveSubmitHandler(e, strm.id )}}>
                                            <button class="btn dark" type="submit">Delete Stream</button>
                                        </form>

                                    </li>
                                </ul>
                            ))
                            }

                        </React.Fragment>

                        <footer style={footerStyle} align="left">
                            
                        </footer>

                    </div>
                    

                </center>
            );
        }
        else if(this.state.errorMessage === '' && this.state.token === null){
            return(
                <div>
                    {console.log("no token branch")}
                    <br/><br/>
                    <h1>You must be logged in to view this page!</h1>
                    {alert("You must be logged in to view this page!")}
                    {window.location.replace("/login")}
                </div>
            );
        }
      
    }
}

export default Video;