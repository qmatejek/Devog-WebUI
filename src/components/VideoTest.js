import React, { Component } from 'react';
import axios from 'axios';

export class VideoTest extends Component {

    state = 
    {
        url: 'http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/live',
        token: localStorage.getItem('token'),
        auth: 'Token ' + localStorage.getItem('token'),
    }


    submitHandler = e => {
        e.preventDefault()
        console.log(this.state)
        axios.get(this.state.url, {headers: {"authorization":this.state.auth}})
        .then(Response => {
            console.log(Response)
        })
        .catch(error => {
            console.log(error)
        })
    }

    render() {
        return (
            <div>
                <form onSubmit={this.submitHandler}>
                    <h1>Press button for Videos</h1>
                    <h3>{this.state.token}</h3>
                    <h3>{this.state.auth}</h3>
                    <button type="submit">Submit</button>
                </form>
                

            </div>
        );
    }
}

export default VideoTest;