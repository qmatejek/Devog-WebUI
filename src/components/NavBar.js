import React, { Component } from 'react';
import { Logout } from './Logout';
import '../index.css';



export class NavBar extends Component {

    state = {
        token: localStorage.getItem('token'),
    }

    

    render(){

        //This piece of code with the button variable allows me to do conditional rendering of the login/logout buttons and the videos link.
        //If token is null, then user is not logged in and needs to see the login button. If there is a token, then he/she needs the logout.
        let button;
        let vidLink;

        if(this.state.token == null){
            button = <span className="navbar-button"><a className="btn dark" href="/login" >login</a></span>;
            vidLink = "";
        }
        else{
            button = <Logout/>;
            vidLink = <span className="navbar-item"><a href='/video'>Videos</a></span>
        }

        

        return(
            
                <div className="navbar-content">
                    <span className="navbar-logo"><a href="/">Home</a></span>
                    {button}
                    <span className="navbar-item"><a href="">About</a></span>
                    <span className="navbar-item"><a href="">Contact</a></span>
                    <span className="navbar-item"><a href="">Devog</a></span>
                    {vidLink}
                </div>
            
        );
    }
}