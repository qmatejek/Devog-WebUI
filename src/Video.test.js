import { Video } from './components/Video';
import { Login } from './Login';
import React from 'react';
import { shallow, render, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import { MemoryRouter } from 'react-router-dom';
import axios from 'axios';
import { cleanup, fireEvent } from '@testing-library/react';
import { Redirect } from "react-router-dom";
import { MemoryRouter as Router } from 'react-router-dom';
configure({ adapter: new Adapter() });

let sandbox;
jest.mock('axios');

beforeEach(() => {
  sandbox = sinon.createSandbox();
  window.alert = sandbox.stub();
  
  React.createRef = sandbox.stub()
    .returns({ current: { setAttribute: sandbox.stub() }});
});

afterEach(() => sandbox.restore());

it("renders correctly", () => {
    const wrapper = shallow(
      <Video history={[]} />
    );
  
    expect(wrapper).toMatchSnapshot();
  });
 

it("fetched live streams using fetchLvSubmitHandler", () =>{
  axios.get.mockImplementationOnce(() => Promise.resolve(''));

  const component = new Video();
  
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const event = { preventDefault() {}, stopPropagation() {} };

  component.setState({auth: "auth123"});

  localStorage.setItem('username', 'sdp');
  component.fetchLvSubmitHandler(event);
    

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/live";
  const headers = {headers: {"authorization":"auth123"}};

  expect(axios.get).toHaveBeenCalledWith(url, headers);
})

it("fetched uploaded videos using fetchUpSubmitHandler", () =>{

  const data = ['1', '2', '3'];

  axios.get.mockImplementationOnce(() => Promise.resolve(data));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const event = { preventDefault() {}, stopPropagation() {} };

  component.setState({auth: "auth123"});

  localStorage.setItem('username', 'sdp');

  component.fetchUpSubmitHandler(event);
  

  let url = 'http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp';
  const headers = {headers: {"authorization":"auth123"}};

  expect(axios.get).toHaveBeenCalledWith(url, headers);
});

 it('fetchUpSubmitHandler sets error message on failure', () => {
   axios.get.mockImplementationOnce(() => 
     new Promise((resolve, reject) => {
       throw new Error('great snakes');
     }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const event = { preventDefault() {}, stopPropagation() {} };
  component.fetchUpSubmitHandler(event)

   return new Promise((resolve, reject) => {
     setImmediate(() => {
       expect(component.state.errorMessage).toBe('great snakes');
       resolve();
     });
   });
 });

it('fetchLvSubmitHandler sets error message on failure', () => {
  axios.get.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const event = { preventDefault() {}, stopPropagation() {} };

  component.fetchLvSubmitHandler(event);

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });
});

it("editSubmitHandler makes API call", ()=> {
  axios.put.mockImplementationOnce(() => Promise.resolve(''));

  sinon.stub(window.location, 'reload');

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}, target:[{value:"test"},]};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"})

  component.editSubmitHandler(e, "00");

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/00";


  expect(axios.put).toHaveBeenCalledWith(url, {"name":"test"}, {headers: {"Authorization":"auth123"}});

})

it("editSubmitHandler sets error message on failure", ()=> {
  axios.put.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}, target:[{value:"test"},]};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"})

  component.editSubmitHandler(e, "00");

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });

})

it("editLiveSubmitHandler makes API call", ()=> {
  axios.put.mockImplementationOnce(() => Promise.resolve(''));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}, target:[{value:"test"},]};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"})

  component.editLiveSubmitHandler(e, "00");

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/live/00";

  expect(axios.put).toHaveBeenCalledWith(url, {"name":"test"}, {headers: {"Authorization":"auth123"}});

});

it("editLiveSubmitHandler sets errorMessage on failure", ()=> {
  axios.put.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}, target:[{value:"test"},]};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"})

  component.editLiveSubmitHandler(e, "00");

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });

});

it("delSubmitHandler makes API call", ()=>{
  axios.delete.mockImplementationOnce(() => Promise.resolve(''));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"})

  component.delSubmitHandler(e, "00");

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/00";

  expect(axios.delete).toHaveBeenCalledWith(url, {headers: {"Authorization":"auth123"}});
})

it("delSubmitHandler sets errorMessage on failure", ()=>{
  axios.delete.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"})

  component.delSubmitHandler(e, "00");

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });
})

it("delLiveSubmitHandler makes API call", ()=>{
  axios.delete.mockImplementationOnce(() => Promise.resolve(''));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"});

  localStorage.setItem('username','sdp');

  component.delLiveSubmitHandler(e, "00");

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/live/00";

  expect(axios.delete).toHaveBeenCalledWith(url, {headers: {"Authorization":"auth123"}});
});

it("delLiveSubmitHandler sets errorMessage on failure", ()=>{
  axios.delete.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}};

  localStorage.setItem('username','sdp');

  component.setState({auth:"auth123"});

  localStorage.setItem('username','sdp');

  component.delLiveSubmitHandler(e, "00");

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });
});

it("addLiveSubmitHandler makes API call", ()=>{
  axios.post.mockImplementationOnce(() => Promise.resolve(''));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}};

  component.setState({streamName: "test", streamURL: "testURL", auth:"auth123"});

  localStorage.setItem('username','sdp');
  component.addLiveSubmitHandler(e);

  const formData = new FormData();
        formData.append('name','test');
        formData.append('link','testURL')

  const headers = {headers: {"Authorization":"auth123", 'Content-Type':'application/json'}};      

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp/live";

  expect(axios.post).toHaveBeenCalledWith(url, formData, headers);
});

it("addLiveSubmitHandler sets errorMessage on failure", ()=>{
  axios.post.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}};

  component.setState({streamName: "test", streamURL: "testURL", auth:"auth123"});

  localStorage.setItem('username','sdp');
  component.addLiveSubmitHandler(e);

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });
});

it("fileButtonHandler makes API call", () =>{
  axios.post.mockImplementationOnce(() => Promise.resolve(''));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}, target:{files:[{name: "test"}]}};


  component.setState({auth:"auth123"});

  localStorage.setItem('username','sdp')
  component.fileButtonHandler(e);

  const formData = new FormData();
        formData.append('name', e.target.files[0].name);
        formData.append('location', e.target.files[0]);

  const headers = {headers: {"Authorization":"auth123", 'Content-Type':'application/json'}};      

  let url = "http://vision-flow-apiapp.ysy9jamj5c.us-east-1.elasticbeanstalk.com/appview/sdp";

  expect(axios.post).toHaveBeenCalledWith(url, formData, headers);
});

it("fileButtonHandler sets errorMessage on failure", () =>{
  axios.post.mockImplementationOnce(() => 
    new Promise((resolve, reject) => {
      throw new Error('great snakes');
    }));

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {preventDefault() {}, stopPropagation() {}, target:{files:[{name: "test"}]}};


  component.setState({auth:"auth123"});

  localStorage.setItem('username','sdp')
  component.fileButtonHandler(e);

  return new Promise((resolve, reject) => {
    setImmediate(() => {
      expect(component.state.errorMessage).toBe('great snakes');
      resolve();
    });
  });
});

it("changeHandler handles change", () => {

  const component = new Video();
  component.setState = function(state) {
    this.state = {...this.state, ...state};
  }

  const e = {target:{name: "streamName", value: "test"}};

  component.changeHandler(e);

  expect(component.state.streamName).toBe("test");
});

it('render error message if there is an error', () => {
  sinon.stub(window.location, 'replace');

  const wrapper = mount(<Video />);
  
  wrapper.instance().setState({ errorMessage: '...details of error...' });

  wrapper.update();

  expect(wrapper.text())
    .toEqual(expect.stringContaining('Oops! An error has occured!'));

  expect(wrapper.text())
    .toEqual(expect.stringContaining('Error: ...details of error...'));
});

// it('render failed to render message if there is nothing to render', () => {
//   const wrapper = shallow(<Video />);

//   wrapper.setState({token: "auth123"})

//   expect(wrapper.text())
//     .toEqual(expect.stringContaining('Oops! Nothing was rendered...'));
// });

it('render not logged in if there is an error', () => {
  //sinon.stub(window.location, 'replace');

  const wrapper = mount(<Video />);

  expect(wrapper.text())
    .toEqual(expect.stringContaining('You must be logged in to view this page!'));

});

// it('fetchUpSubmitHandler fetches videos', () =>{
//   const wrapper = mount(<Video />);

//   wrapper.find("button")
//         .first()
//         .simulate("click")

//   expect(wrapper.state("videos")).not.toBe(null);      
// });