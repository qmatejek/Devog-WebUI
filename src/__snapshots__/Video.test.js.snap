// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <br />
  <br />
  <h1>
    You must be logged in to view this page!
  </h1>
</div>
`;
